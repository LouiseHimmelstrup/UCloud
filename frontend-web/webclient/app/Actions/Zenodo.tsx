import { Cloud } from "../../authentication/SDUCloudObject";
import {
    RECEIVE_PUBLICATIONS,
    SET_ZENODO_LOADING
} from "../Reducers/Zenodo";
import { SetLoadingAction, ReceivePage } from "../types/types";

export const fetchPublications = (page: number, itemsPerPage: number): Promise<ReceivePublications> =>
    Cloud.get(`/zenodo/publications/?itemsPerPage=${itemsPerPage}&page=${page}`).then(({ response }) => {
        return receivePublications(response.inProgress, response.connected);
    }).catch(failure => {
        return receivePublications([], false);
    });

interface ReceivePublications extends ReceivePage<File> { connected: boolean }
const receivePublications = (page, connected): ReceivePublications => ({
    type: RECEIVE_PUBLICATIONS,
    connected,
    page
})

export const setZenodoLoading = (loading): SetLoadingAction => ({
    type: SET_ZENODO_LOADING,
    loading
});