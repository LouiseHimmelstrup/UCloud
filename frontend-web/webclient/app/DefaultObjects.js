import {Cloud} from "../authentication/SDUCloudObject"
import {tusConfig} from "./Configurations";
import Uppy from "uppy";

export const DefaultStatus = {
    title: "No Issues",
    level: "NO ISSUES",
    body: "The system is running as intended.",
};

export const RightsMap = {
    NONE: 0,
    READ: 1,
    READ_WRITE: 2,
    OWN: 3,
};

export const AnalysesStatusMap = {
    "PENDING": 0,
    "IN PROGRESS": 1,
    "COMPLETED": 2,
};

export const RightsNameMap = {
    NONE: "None",
    READ: "Read",
    READ_WRITE: "Read/Write",
    OWN: "Own",
};

export const SensitivityLevel = {
    "OPEN_ACCESS": "Open Access",
    "CONFIDENTIAL": "Confidential",
    "SENSITIVE": "Sensitive",
};

export const SensitivityLevelMap = {
    "OPEN_ACCESS": 0,
    "CONFIDENTIAL": 1,
    "SENSITIVE": 2,
};

const initializeUppy = (restrictions) => 
    Uppy.Core({
        autoProceed: false,
        debug: false,
        restrictions: restrictions,
        meta: {
            sensitive: false,
        },
        onBeforeUpload: () => {
            return Cloud.receiveAccessTokenOrRefreshIt().then((data) => {
                tusConfig.headers["Authorization"] = `Bearer ${data}`;
            });
        }
    }).use(Uppy.Tus, tusConfig);

export const initObject = {
    files: {
        files: [],
        filesShown: 10,
        filesPerPage: 10,
        currentFilesPage: 0,
        loading: false,
        path: "",
        filesInfoPath: "",
        filesInfoFile: null,
        projects: []
    },
    uppy: {
        uppyFiles: initializeUppy({maxNumberOfFiles: false}),
        uppyFilesOpen: false,
        uppyRunApp: initializeUppy({ maxNumberOfFiles: 1}),
        uppyRunAppOpen: false
    },
    status: {
        status: DefaultStatus,
        title: ""
    }
};