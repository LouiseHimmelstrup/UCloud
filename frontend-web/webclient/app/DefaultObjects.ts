import {SidebarOption, Page} from "Types";
import {Status} from "Navigation";
import {Analysis, DetailedApplicationSearchReduxState, RunsSortBy} from "Applications";
import {File, DetailedFileSearchReduxState} from "Files";
import {SortOrder, SortBy} from "Files";
import {DashboardStateProps} from "Dashboard";
import {Publication} from "Zenodo";
import {Notification} from "Notifications";
import {Upload} from "Uploader";
import {Activity, ActivityGroup, ActivityFilter} from "Activity";
import {Reducer} from "redux";
import {SimpleSearchStateProps} from "Search";
import * as ApplicationRedux from "Applications/Redux";
import * as AccountingRedux from "Accounting/Redux";
import * as SnackbarRedux from "Snackbar/Redux";
import * as FavoritesRedux from "Favorites/Redux";
import {defaultAvatar} from "UserSettings/Avataaar";
import {SidebarPages} from "ui-components/Sidebar";
import {ScrollResult} from "Scroll/Types";
import * as ProjectRedux from "Project/Redux";

export const DefaultStatus: Status = {
    title: "No Issues",
    level: "NO ISSUES",
    body: "The system is running as intended."
};

export enum KeyCode {
    ENTER = 13,
    ESC = 27
}

export const emptyPage: Page<any> = {items: [], itemsPerPage: 25, itemsInTotal: 0, pageNumber: 0, pagesInTotal: 0};

export enum AnalysesStatusMap {
    "PENDING",
    "IN PROGRESS",
    "COMPLETED"
}

export enum RightsNameMap {
    "NONE" = "None",
    "READ" = "Read",
    "READ_WRITE" = "Read/Write"
}

export enum SensitivityLevel {
    "INHERIT" = "Inherit",
    "PRIVATE" = "Private",
    "CONFIDENTIAL" = "Confidential",
    "SENSITIVE" = "Sensitive"
}

export type Sensitivity = keyof typeof SensitivityLevel;

export enum SensitivityLevelMap {
    INHERIT = "INHERIT",
    PRIVATE = "PRIVATE",
    CONFIDENTIAL = "CONFIDENTIAL",
    SENSITIVE = "SENSITIVE"
}

function getFilesSortingColumnOrDefault(columnIndex: 0 | 1): SortBy {
    const sortingColumn = window.localStorage.getItem(`filesSorting${columnIndex}`);
    if (sortingColumn && Object.values(SortBy).includes(sortingColumn)) return sortingColumn as SortBy;
    switch (columnIndex) {
        case 0:
            window.localStorage.setItem("filesSorting0", SortBy.MODIFIED_AT);
            return SortBy.MODIFIED_AT;
        case 1:
            window.localStorage.setItem("filesSorting1", SortBy.SIZE);
            return SortBy.SIZE;
    }
}

function getItemOrDefault<T, T2>(itemName: string, defaultValue: T, en: T2): T {
    const item = window.localStorage.getItem(itemName);
    if (item && Object.values(en).includes(item)) {
        return item as unknown as T;
    }
    return defaultValue;
}

export interface ComponentWithLoadingState {
    loading: boolean
    error?: string
}

export interface ComponentWithPage<T> extends ComponentWithLoadingState {
    page: Page<T>
}

export interface ComponentWithScroll<Item, OffsetType> extends ComponentWithLoadingState {
    scroll?: ScrollResult<Item, OffsetType>
}

export interface ResponsiveReduxObject {
    mediaType: string
    orientation: string
    lessThan: Record<string, boolean>
    greaterThan: Record<string, boolean>
    is: Record<string, boolean>
}

export const initResponsive = (): ResponsiveReduxObject => ({
    mediaType: "",
    orientation: "",
    lessThan: {},
    greaterThan: {},
    is: {},
});

export interface FilePreviewReduxState {
    file?: File
    error?: string
}

export interface FilesReduxObject extends ComponentWithPage<File> {
    sortOrder: SortOrder
    sortBy: SortBy
    path: string
    filesInfoPath: string
    fileSelectorError?: string
    sortingColumns: [SortBy, SortBy]
    fileSelectorLoading: boolean
    fileSelectorShown: boolean
    fileSelectorPage: Page<File>
    fileSelectorPath: string
    fileSelectorCallback: Function
    fileSelectorIsFavorites: boolean
    disallowedPaths: string[]
    invalidPath: boolean
}

export interface FileInfoReduxObject {
    file?: File
    error?: string
    activity: Page<Activity>
    loading: boolean
}

export interface AnalysisReduxObject extends ComponentWithPage<Analysis> {
    sortBy: RunsSortBy
    sortOrder: SortOrder
}

export interface NotificationsReduxObject {
    redirectTo: string
    items: Notification[]
    loading: boolean
    error?: string
}

export interface ZenodoReduxObject extends ComponentWithPage<Publication> {
    connected: boolean
}

export interface StatusReduxObject {
    status: Status
    title: string
    page: SidebarPages
    loading: boolean
}

export interface SidebarReduxObject {
    pp: boolean
    options: SidebarOption[]
    kcCount: number
}

export interface HeaderSearchReduxObject {
    prioritizedSearch: HeaderSearchType
    refresh?: () => void
}

export interface RunApplicationReduxObject {

}

export type ActivityReduxObject = ComponentWithScroll<ActivityGroup, number> & ActivityFilter;

export type HeaderSearchType = "files" | "applications" | "projects";

export interface UploaderReduxObject {
    uploads: Upload[]
    visible: boolean
    path: string
    allowMultiple: boolean
    onFilesUploaded: (p: string) => void
    error?: string
    loading: boolean
}

interface LegacyReducers {
    dashboard?: Reducer<DashboardStateProps>
    files?: Reducer<FilesReduxObject>
    uploader?: Reducer<UploaderReduxObject>
    status?: Reducer<StatusReduxObject>
    notifications?: Reducer<NotificationsReduxObject>
    analyses?: Reducer<AnalysisReduxObject>
    zenodo?: Reducer<ZenodoReduxObject>
    header?: Reducer<HeaderSearchReduxObject>
    sidebar?: Reducer<SidebarReduxObject>
    activity?: Reducer<ActivityReduxObject>
    detailedResult?: Reducer<DetailedResultReduxObject>
}

export type Reducers = LegacyReducers & ApplicationRedux.Reducers & AccountingRedux.Reducers;

export type DetailedResultReduxObject = ComponentWithPage<File>

export const initDetailedResult = (): DetailedResultReduxObject => ({
    page: emptyPage,
    loading: false,
    error: undefined
});

/* FIXME */
interface LegacyReduxObject {
    dashboard: DashboardStateProps
    files: FilesReduxObject,
    uploader: UploaderReduxObject
    status: StatusReduxObject,
    notifications: NotificationsReduxObject
    analyses: AnalysisReduxObject
    zenodo: ZenodoReduxObject
    header: HeaderSearchReduxObject
    sidebar: SidebarReduxObject
    activity: ActivityReduxObject
    detailedResult: DetailedResultReduxObject
    simpleSearch: SimpleSearchStateProps
    detailedFileSearch: DetailedFileSearchReduxState
    detailedApplicationSearch: DetailedApplicationSearchReduxState
    fileInfo: FileInfoReduxObject
    avatar: AvatarReduxObject
    filePreview: FilePreviewReduxState
    responsive?: ResponsiveReduxObject
    project: ProjectRedux.State
    loading?: boolean
}

export type ReduxObject =
    LegacyReduxObject &
    ApplicationRedux.Objects &
    AccountingRedux.Objects &
    FavoritesRedux.Objects &
    SnackbarRedux.Wrapper;

export const initActivity = (): ActivityReduxObject => ({
    loading: false
});

export const initNotifications = (): NotificationsReduxObject => ({
    items: [],
    loading: false,
    redirectTo: "",
    error: undefined
});

export const initHeader = (): HeaderSearchReduxObject => ({
    prioritizedSearch: "files"
});

export const initStatus = (): StatusReduxObject => ({
    status: DefaultStatus,
    title: "",
    page: SidebarPages.None,
    loading: false
});

export const initDashboard = (): DashboardStateProps => ({
    favoriteFiles: [],
    recentFiles: [],
    recentAnalyses: [],
    notifications: [],
    favoriteLoading: false,
    recentLoading: false,
    analysesLoading: false
});

export function initObject(homeFolder: string): ReduxObject {
    return {
        dashboard: initDashboard(),
        files: initFiles(homeFolder),
        status: initStatus(),
        header: initHeader(),
        notifications: initNotifications(),
        analyses: initAnalyses(),
        zenodo: initZenodo(),
        sidebar: initSidebar(),
        uploader: initUploads(),
        activity: initActivity(),
        detailedResult: initDetailedResult(),
        simpleSearch: initSimpleSearch(),
        detailedApplicationSearch: initApplicationsAdvancedSearch(),
        detailedFileSearch: initFilesDetailedSearch(),
        fileInfo: initFileInfo(),
        avatar: initAvatar(),
        filePreview: initFilePreview(),
        project: ProjectRedux.initialState,
        ...ApplicationRedux.init(),
        ...AccountingRedux.init(),
        ...FavoritesRedux.init(),
        ...SnackbarRedux.init(),
        responsive: undefined,
    }
}


export const initFilePreview = () => ({
    file: undefined,
    error: undefined
});

export type AvatarReduxObject = typeof defaultAvatar & { error?: string };
export const initAvatar = (): AvatarReduxObject => ({...defaultAvatar, error: undefined});

export const initSimpleSearch = (): SimpleSearchStateProps => ({
    files: emptyPage,
    filesLoading: false,
    applications: emptyPage,
    applicationsLoading: false,
    errors: [],
    search: "",
    applicationSearch: initApplicationsAdvancedSearch(),
    fileSearch: initFilesDetailedSearch()
});

export const initAnalyses = (): AnalysisReduxObject => ({
    page: emptyPage,
    loading: false,
    error: undefined,
    sortBy: RunsSortBy.createdAt,
    sortOrder: SortOrder.ASCENDING
});

export const initZenodo = (): ZenodoReduxObject => ({
    connected: false,
    loading: false,
    page: emptyPage
});

export const initSidebar = (): SidebarReduxObject => ({
    pp: false,
    kcCount: 0,
    options: []
});

export const initUploads = (): UploaderReduxObject => ({
    path: "",
    uploads: [],
    visible: false,
    allowMultiple: false,
    error: undefined,
    onFilesUploaded: () => null,
    loading: false
});

export const initFileInfo = (): FileInfoReduxObject => ({
    activity: emptyPage,
    loading: false
});

export const initFiles = (homeFolder: string): FilesReduxObject => ({
    page: emptyPage,
    sortOrder: getItemOrDefault("sortOrder", SortOrder.ASCENDING, SortOrder),
    sortBy: getItemOrDefault("sortBy", SortBy.PATH, SortBy),
    loading: false,
    error: undefined,
    path: "",
    filesInfoPath: "",
    sortingColumns: [getFilesSortingColumnOrDefault(0), getFilesSortingColumnOrDefault(1)],
    fileSelectorLoading: false,
    fileSelectorShown: false,
    fileSelectorPage: emptyPage,
    fileSelectorPath: homeFolder,
    fileSelectorIsFavorites: false,
    fileSelectorCallback: () => undefined,
    fileSelectorError: undefined,
    disallowedPaths: [],
    invalidPath: false
});

export const initFilesDetailedSearch = (): DetailedFileSearchReduxState => ({
    hidden: true,
    allowFolders: true,
    allowFiles: true,
    fileName: "",
    extensions: new Set(),
    tags: new Set(),
    sensitivities: new Set(),
    createdBefore: undefined,
    createdAfter: undefined,
    modifiedBefore: undefined,
    modifiedAfter: undefined,
    error: undefined,
    loading: false
});

export const initApplicationsAdvancedSearch = (): DetailedApplicationSearchReduxState => ({
    error: undefined,
    loading: false,
    hidden: true,
    appName: "",
    appVersion: "",
    tags: ""
});
