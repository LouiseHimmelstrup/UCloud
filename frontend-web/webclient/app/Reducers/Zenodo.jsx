export const SET_LOADING = "SET_LOADING";
export const RECEIVE_PUBLICATIONS = "RECEIVE_PUBLICATIONS";

const zenodo = (state = [], action) => {
    switch (action.type) {
        case RECEIVE_PUBLICATIONS: {
            return { ...state, publications: action.publications, connected: action.connected, loading: false };
        }
        case SET_LOADING: {
            return { ...state, loading: action.loading };
        }
        default: {
            return state;
        }
    }
}

export default zenodo;