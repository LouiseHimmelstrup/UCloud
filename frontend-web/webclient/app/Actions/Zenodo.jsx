import { Cloud } from "../../authentication/SDUCloudObject";
import { RECEIVE_PUBLICATIONS, SET_LOADING } from "../Reducers/Zenodo";

export const fetchPublications = () =>
    Cloud.get("/zenodo/publications").then(({ response }) => {
        return receivePublications(response.inProgress, response.connected);
    }).catch(failure => {
        return receivePublications([], false);
    });

const receivePublications = (publications, connected) => ({
    type: RECEIVE_PUBLICATIONS,
    publications,
    connected
})

export const setLoading = (loading) => ({
    type: SET_LOADING,
    loading
});