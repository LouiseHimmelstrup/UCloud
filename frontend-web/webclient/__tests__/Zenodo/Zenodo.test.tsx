import Zenodo from "Zenodo/Zenodo";
import { create } from "react-test-renderer";
import { MemoryRouter } from "react-router";
import { configureStore } from "Utilities/ReduxUtilities";
import { Provider } from "react-redux";
import * as React from "react";
import zenodo from "Zenodo/Redux/ZenodoReducer";
import { initZenodo } from "DefaultObjects";


describe("Zenodo", () => {
    test("Mount Zenodo component", () => {
        expect(create(
            <Provider store={configureStore({ zenodo: initZenodo() }, { zenodo })}>
                <MemoryRouter>
                    <Zenodo />
                </MemoryRouter>
            </Provider>
        ).toJSON()).toMatchSnapshot()
    });
});