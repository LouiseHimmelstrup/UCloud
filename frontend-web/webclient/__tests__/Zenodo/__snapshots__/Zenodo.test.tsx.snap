// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Zenodo Mount Zenodo component 1`] = `
Array [
  <div
    className="sc-bdVaJa guSZfy"
  >
    <h2
      className="sc-EHOje cMXCEK"
      cursor="inherit"
      fontSize={5}
    >
      Upload progress
    </h2>
    <h5
      className="sc-EHOje hviljP"
      cursor="inherit"
      fontSize={2}
    >
      Connected to Zenodo
    </h5>
  </div>,
  <div
    className="sc-ifAKCX iCcktw"
  >
    <div
      className="sc-bdVaJa bXFxSh"
      width={
        Array [
          1,
          0.8125,
        ]
      }
    >
      <div
        className="sc-fjdhpX llNXsf"
      >
        <a
          className="sc-jKJlTe cboMKF"
          color="text"
          hoverColor="textHighlight"
          href="/zenodo/publish/"
          onClick={[Function]}
          theme={
            Object {
              "bold": 700,
              "boxShadows": Array [
                "0 0 2px 0 rgba(0,0,0,.08),0 1px 4px 0 rgba(0,0,0,.16)",
                "0 0 2px 0 rgba(0,0,0,.08),0 2px 8px 0 rgba(0,0,0,.16)",
                "0 0 2px 0 rgba(0,0,0,.08),0 4px 16px 0 rgba(0,0,0,.16)",
                "0 0 2px 0 rgba(0,0,0,.08),0 8px 32px 0 rgba(0,0,0,.16)",
              ],
              "breakpoints": Array [
                "32em",
                "40em",
                "48em",
                "64em",
              ],
              "colorStyles": Object {
                "blueOnLightBlue": Object {
                  "backgroundColor": "#cdf",
                  "color": "#0055d5",
                },
                "blueOnWhite": Object {
                  "backgroundColor": "#fff",
                  "color": "#0055d5",
                },
                "danger": Object {
                  "backgroundColor": "#c00",
                  "color": "#fff",
                },
                "grayOnWhite": Object {
                  "backgroundColor": "#fff",
                  "color": "#8393A7",
                },
                "greenOnLightGreen": Object {
                  "backgroundColor": "#cec",
                  "color": "#0a0",
                },
                "greenOnWhite": Object {
                  "backgroundColor": "#fff",
                  "color": "#0a0",
                },
                "info": Object {
                  "backgroundColor": "#ebeff3",
                  "color": "#53657d",
                },
                "purpleOnLightPurple": Object {
                  "backgroundColor": "#ecf",
                  "color": "#70b",
                },
                "purpleOnWhite": Object {
                  "backgroundColor": "#fff",
                  "color": "#70b",
                },
                "redOnLightRed": Object {
                  "backgroundColor": "#fcc",
                  "color": "#c00",
                },
                "redOnWhite": Object {
                  "backgroundColor": "#fff",
                  "color": "#c00",
                },
                "success": Object {
                  "backgroundColor": "#0a0",
                  "color": "#fff",
                },
                "textOnLightGray": Object {
                  "backgroundColor": "#ebeff3",
                  "color": "#53657d",
                },
                "textOnOrange": Object {
                  "backgroundColor": "#fa0",
                  "color": "#53657d",
                },
                "textOnWhite": Object {
                  "backgroundColor": "#fff",
                  "color": "#53657d",
                },
                "warning": Object {
                  "backgroundColor": "#fa0",
                  "color": "#53657d",
                },
                "whiteOnBlue": Object {
                  "backgroundColor": "#0055d5",
                  "color": "#fff",
                },
                "whiteOnDarkOrange": Object {
                  "backgroundColor": "#a50",
                  "color": "#fff",
                },
                "whiteOnGray": Object {
                  "backgroundColor": "#8393A7",
                  "color": "#fff",
                },
                "whiteOnGreen": Object {
                  "backgroundColor": "#0a0",
                  "color": "#fff",
                },
                "whiteOnPurple": Object {
                  "backgroundColor": "#70b",
                  "color": "#fff",
                },
                "whiteOnRed": Object {
                  "backgroundColor": "#c00",
                  "color": "#fff",
                },
                "whiteOnText": Object {
                  "backgroundColor": "#53657d",
                  "color": "#fff",
                },
              },
              "colors": Object {
                "black": "#000",
                "blue": "#0055d5",
                "borderGray": "#c9d3df",
                "darkBlue": "#049",
                "darkGray": "#53657d",
                "darkGreen": "#060",
                "darkOrange": "#a50",
                "darkRed": "#800",
                "gray": "#8393A7",
                "green": "#0a0",
                "headerBg": "#0055d5",
                "headerIconColor": "#ebeff3",
                "headerIconColor2": "#c9d3df",
                "headerText": "#ebeff3",
                "iconColor": "#53657d",
                "iconColor2": "#8393A7",
                "lightBlue": "#cdf",
                "lightGray": "#ebeff3",
                "lightGreen": "#cec",
                "lightPurple": "#ecf",
                "lightRed": "#fcc",
                "midGray": "#c9d3df",
                "orange": "#fa0",
                "paginationDisabled": "#ebeff3",
                "paginationHoverColor": "#cdf",
                "purple": "#70b",
                "red": "#c00",
                "text": "#53657d",
                "textHighlight": "#0055d5",
                "white": "#fff",
              },
              "duration": Object {
                "fast": "150ms",
                "normal": "300ms",
                "slow": "450ms",
                "slowest": "600ms",
              },
              "fontFamily": "'IBM Plex Sans', sans-serif",
              "fontSizes": Array [
                12,
                14,
                16,
                20,
                24,
                32,
                40,
                56,
                72,
              ],
              "fontWeights": Object {
                "bold": 700,
                "medium": 300,
                "regular": 400,
              },
              "letterSpacings": Object {
                "caps": "0.025em",
                "normal": "normal",
              },
              "lineHeights": Object {
                "display": 1.25,
                "standard": 1.5,
              },
              "maxContainerWidth": "1280px",
              "mediaQueries": Array [
                "@media screen and (min-width:32em)",
                "@media screen and (min-width:40em)",
                "@media screen and (min-width:48em)",
                "@media screen and (min-width:64em)",
              ],
              "radii": Array [
                0,
                2,
                6,
              ],
              "radius": "2px",
              "regular": 400,
              "space": Array [
                0,
                4,
                8,
                16,
                32,
                64,
                128,
              ],
              "textStyles": Object {
                "body0": Object {
                  "fontSize": "12px",
                  "fontWeight": 300,
                  "lineHeight": 1.5,
                },
                "body1": Object {
                  "fontSize": "14px",
                  "fontWeight": 300,
                  "lineHeight": 1.5,
                },
                "body2": Object {
                  "fontSize": "16px",
                  "fontWeight": 300,
                  "lineHeight": 1.5,
                },
                "display0": Object {
                  "fontSize": "12px",
                  "fontWeight": 700,
                  "letterSpacing": "0.025em",
                  "lineHeight": 1.25,
                  "textTransform": "uppercase",
                },
                "display1": Object {
                  "fontSize": "14px",
                  "fontWeight": 700,
                  "lineHeight": 1.25,
                },
                "display2": Object {
                  "fontSize": "16px",
                  "fontWeight": 700,
                  "lineHeight": 1.25,
                },
                "display3": Object {
                  "fontSize": "20px",
                  "fontWeight": 700,
                  "lineHeight": 1.25,
                },
                "display4": Object {
                  "fontSize": "24px",
                  "fontWeight": 700,
                  "lineHeight": 1.25,
                },
                "display5": Object {
                  "fontSize": "32px",
                  "fontWeight": 700,
                  "lineHeight": 1.25,
                },
                "display6": Object {
                  "fontSize": "40px",
                  "fontWeight": 700,
                  "lineHeight": 1.25,
                },
                "display7": Object {
                  "fontSize": "56px",
                  "fontWeight": 700,
                  "lineHeight": 1.25,
                },
                "display8": Object {
                  "fontSize": "72px",
                  "fontWeight": 700,
                  "lineHeight": 1.25,
                },
              },
              "timingFunctions": Object {
                "easeIn": "cubic-bezier(0.5, 0, 1, 1)",
                "easeInOut": "cubic-bezier(0.5, 0, 0.25, 1)",
                "easeOut": "cubic-bezier(0, 0, 0.25, 1)",
              },
              "transitionDelays": Object {
                "large": "260ms",
                "medium": "160ms",
                "small": "60ms",
                "xLarge": "360ms",
              },
            }
          }
        >
          <button
            className="sc-eNQAEJ fQogFc"
            color="blue"
          >
            Create new upload
          </button>
        </a>
      </div>
      <div
        className="sc-ifAKCX dzFYJo"
      >
        <div
          className="sc-bdVaJa KmLFy"
        />
        <div
          className="sc-hSdWYo hKqwIs"
        >
          <div
            className="sc-gipzik iadCeV"
          >
            <span
              className="sc-EHOje gynpeM"
              cursor="pointer"
              onClick={[Function]}
            >
              <div
                className="sc-ifAKCX hEFUra"
              >
                <div
                  className="sc-bdVaJa kCeIih"
                >
                   
                  Items per page 25
                </div>
                <div
                  className="sc-bdVaJa kCeIih"
                >
                  <svg
                    className="sc-gzVnrw dXwGzW"
                    cursor="auto"
                    fill="currentcolor"
                    height={24}
                    viewBox="0 0 24 24"
                    width={24}
                  >
                    <path
                      d="M7.4 8l4.6 4.6L16.6 8 18 9.4l-6 6-6-6L7.4 8z"
                    />
                  </svg>
                </div>
              </div>
            </span>
          </div>
          <i
            className="fas fa-sync fa-spin"
            onClick={[Function]}
          />
        </div>
      </div>
      <div
        className="sc-ifAKCX hEFUra"
      >
        <div
          className="sc-bdVaJa jhLdJF"
          width="50%"
        />
        <i
          className="fas fa-circle-notch fa-spin"
        />
      </div>
      <div />
    </div>
    <div
      className="sc-fjdhpX bqztPG"
      width={
        Array [
          0,
          0.1875,
        ]
      }
    >
      <div
        className="sc-bdVaJa hdnbQo"
      >
        <a
          className="sc-jKJlTe cboMKF"
          color="text"
          hoverColor="textHighlight"
          href="/zenodo/publish/"
          onClick={[Function]}
          theme={
            Object {
              "bold": 700,
              "boxShadows": Array [
                "0 0 2px 0 rgba(0,0,0,.08),0 1px 4px 0 rgba(0,0,0,.16)",
                "0 0 2px 0 rgba(0,0,0,.08),0 2px 8px 0 rgba(0,0,0,.16)",
                "0 0 2px 0 rgba(0,0,0,.08),0 4px 16px 0 rgba(0,0,0,.16)",
                "0 0 2px 0 rgba(0,0,0,.08),0 8px 32px 0 rgba(0,0,0,.16)",
              ],
              "breakpoints": Array [
                "32em",
                "40em",
                "48em",
                "64em",
              ],
              "colorStyles": Object {
                "blueOnLightBlue": Object {
                  "backgroundColor": "#cdf",
                  "color": "#0055d5",
                },
                "blueOnWhite": Object {
                  "backgroundColor": "#fff",
                  "color": "#0055d5",
                },
                "danger": Object {
                  "backgroundColor": "#c00",
                  "color": "#fff",
                },
                "grayOnWhite": Object {
                  "backgroundColor": "#fff",
                  "color": "#8393A7",
                },
                "greenOnLightGreen": Object {
                  "backgroundColor": "#cec",
                  "color": "#0a0",
                },
                "greenOnWhite": Object {
                  "backgroundColor": "#fff",
                  "color": "#0a0",
                },
                "info": Object {
                  "backgroundColor": "#ebeff3",
                  "color": "#53657d",
                },
                "purpleOnLightPurple": Object {
                  "backgroundColor": "#ecf",
                  "color": "#70b",
                },
                "purpleOnWhite": Object {
                  "backgroundColor": "#fff",
                  "color": "#70b",
                },
                "redOnLightRed": Object {
                  "backgroundColor": "#fcc",
                  "color": "#c00",
                },
                "redOnWhite": Object {
                  "backgroundColor": "#fff",
                  "color": "#c00",
                },
                "success": Object {
                  "backgroundColor": "#0a0",
                  "color": "#fff",
                },
                "textOnLightGray": Object {
                  "backgroundColor": "#ebeff3",
                  "color": "#53657d",
                },
                "textOnOrange": Object {
                  "backgroundColor": "#fa0",
                  "color": "#53657d",
                },
                "textOnWhite": Object {
                  "backgroundColor": "#fff",
                  "color": "#53657d",
                },
                "warning": Object {
                  "backgroundColor": "#fa0",
                  "color": "#53657d",
                },
                "whiteOnBlue": Object {
                  "backgroundColor": "#0055d5",
                  "color": "#fff",
                },
                "whiteOnDarkOrange": Object {
                  "backgroundColor": "#a50",
                  "color": "#fff",
                },
                "whiteOnGray": Object {
                  "backgroundColor": "#8393A7",
                  "color": "#fff",
                },
                "whiteOnGreen": Object {
                  "backgroundColor": "#0a0",
                  "color": "#fff",
                },
                "whiteOnPurple": Object {
                  "backgroundColor": "#70b",
                  "color": "#fff",
                },
                "whiteOnRed": Object {
                  "backgroundColor": "#c00",
                  "color": "#fff",
                },
                "whiteOnText": Object {
                  "backgroundColor": "#53657d",
                  "color": "#fff",
                },
              },
              "colors": Object {
                "black": "#000",
                "blue": "#0055d5",
                "borderGray": "#c9d3df",
                "darkBlue": "#049",
                "darkGray": "#53657d",
                "darkGreen": "#060",
                "darkOrange": "#a50",
                "darkRed": "#800",
                "gray": "#8393A7",
                "green": "#0a0",
                "headerBg": "#0055d5",
                "headerIconColor": "#ebeff3",
                "headerIconColor2": "#c9d3df",
                "headerText": "#ebeff3",
                "iconColor": "#53657d",
                "iconColor2": "#8393A7",
                "lightBlue": "#cdf",
                "lightGray": "#ebeff3",
                "lightGreen": "#cec",
                "lightPurple": "#ecf",
                "lightRed": "#fcc",
                "midGray": "#c9d3df",
                "orange": "#fa0",
                "paginationDisabled": "#ebeff3",
                "paginationHoverColor": "#cdf",
                "purple": "#70b",
                "red": "#c00",
                "text": "#53657d",
                "textHighlight": "#0055d5",
                "white": "#fff",
              },
              "duration": Object {
                "fast": "150ms",
                "normal": "300ms",
                "slow": "450ms",
                "slowest": "600ms",
              },
              "fontFamily": "'IBM Plex Sans', sans-serif",
              "fontSizes": Array [
                12,
                14,
                16,
                20,
                24,
                32,
                40,
                56,
                72,
              ],
              "fontWeights": Object {
                "bold": 700,
                "medium": 300,
                "regular": 400,
              },
              "letterSpacings": Object {
                "caps": "0.025em",
                "normal": "normal",
              },
              "lineHeights": Object {
                "display": 1.25,
                "standard": 1.5,
              },
              "maxContainerWidth": "1280px",
              "mediaQueries": Array [
                "@media screen and (min-width:32em)",
                "@media screen and (min-width:40em)",
                "@media screen and (min-width:48em)",
                "@media screen and (min-width:64em)",
              ],
              "radii": Array [
                0,
                2,
                6,
              ],
              "radius": "2px",
              "regular": 400,
              "space": Array [
                0,
                4,
                8,
                16,
                32,
                64,
                128,
              ],
              "textStyles": Object {
                "body0": Object {
                  "fontSize": "12px",
                  "fontWeight": 300,
                  "lineHeight": 1.5,
                },
                "body1": Object {
                  "fontSize": "14px",
                  "fontWeight": 300,
                  "lineHeight": 1.5,
                },
                "body2": Object {
                  "fontSize": "16px",
                  "fontWeight": 300,
                  "lineHeight": 1.5,
                },
                "display0": Object {
                  "fontSize": "12px",
                  "fontWeight": 700,
                  "letterSpacing": "0.025em",
                  "lineHeight": 1.25,
                  "textTransform": "uppercase",
                },
                "display1": Object {
                  "fontSize": "14px",
                  "fontWeight": 700,
                  "lineHeight": 1.25,
                },
                "display2": Object {
                  "fontSize": "16px",
                  "fontWeight": 700,
                  "lineHeight": 1.25,
                },
                "display3": Object {
                  "fontSize": "20px",
                  "fontWeight": 700,
                  "lineHeight": 1.25,
                },
                "display4": Object {
                  "fontSize": "24px",
                  "fontWeight": 700,
                  "lineHeight": 1.25,
                },
                "display5": Object {
                  "fontSize": "32px",
                  "fontWeight": 700,
                  "lineHeight": 1.25,
                },
                "display6": Object {
                  "fontSize": "40px",
                  "fontWeight": 700,
                  "lineHeight": 1.25,
                },
                "display7": Object {
                  "fontSize": "56px",
                  "fontWeight": 700,
                  "lineHeight": 1.25,
                },
                "display8": Object {
                  "fontSize": "72px",
                  "fontWeight": 700,
                  "lineHeight": 1.25,
                },
              },
              "timingFunctions": Object {
                "easeIn": "cubic-bezier(0.5, 0, 1, 1)",
                "easeInOut": "cubic-bezier(0.5, 0, 0.25, 1)",
                "easeOut": "cubic-bezier(0, 0, 0.25, 1)",
              },
              "transitionDelays": Object {
                "large": "260ms",
                "medium": "160ms",
                "small": "60ms",
                "xLarge": "360ms",
              },
            }
          }
        >
          <button
            className="sc-eNQAEJ fQogFc"
            color="blue"
          >
            Create new upload
          </button>
        </a>
      </div>
    </div>
  </div>,
]
`;
